mentat/code_context.py

```diff
@@ -81,188 +75,113 @@ def display_context(self):
 
-        features = None
-        if self.features:
-            stream.send(f"{prefix}Active Features:")
-            features = self.features
-        elif self.include_files:
+        if self.include_files:
             stream.send(f"{prefix}Included files:")
             stream.send(f"{prefix + prefix}{session_context.cwd.name}")
             features = [
-                _feat for _file in self.include_files.values() for _feat in _file
+                feature
+                for file_features in self.include_files.values()
+                for feature in file_features
             ]
+            refs = get_consolidated_feature_refs(features)
+            print_path_tree(
+                build_path_tree([Path(r) for r in refs], session_context.cwd),
+                get_paths_with_git_diffs(self.git_root) if self.git_root else set(),
+                session_context.cwd,
+                prefix + prefix,
+            )
         else:
             stream.send(f"{prefix}Included files: ", end="")
             stream.send("None", color="yellow")
 
-        if features is not None:
-            refs = get_consolidated_feature_refs(features)
+        if self.auto_features:
+            stream.send(f"{prefix}Auto-Included Features:")
+            refs = get_consolidated_feature_refs(self.auto_features)
             print_path_tree(
                 build_path_tree([Path(r) for r in refs], session_context.cwd),
                 get_paths_with_git_diffs(self.git_root) if self.git_root else set(),
                 session_context.cwd,
                 prefix + prefix,
             )
 
-    _code_message: str | None = None
-    _code_message_checksum: str | None = None
-
-    def _get_code_message_checksum(
-        self, prompt: Optional[str] = None, max_tokens: Optional[int] = None
-    ) -> str:
-        session_context = SESSION_CONTEXT.get()
-        config = session_context.config
-        code_file_manager = session_context.code_file_manager
-
-        if not self.features:
-            features_checksum = ""
-        else:
-            feature_files = {
-                session_context.cwd / f.path
-                for f in self.features
-                if (session_context.cwd / f.path).exists()
-            }
-            feature_file_checksums = [
-                code_file_manager.get_file_checksum(f) for f in feature_files
-            ]
-            features_checksum = sha256("".join(feature_file_checksums))
-        settings = {
-            "prompt": prompt or "",
-            "auto_context": config.auto_context,
-            "use_llm": self.use_llm,
-            "diff": self.diff,
-            "pr_diff": self.pr_diff,
-            "max_tokens": max_tokens or "",
-            "include_files": self.include_files,
-        }
-        settings_checksum = sha256(str(settings))
-        return features_checksum + settings_checksum
+    use_llm: bool = False
 
     async def get_code_message(
         self,
+        prompt_tokens: int,
```

Comment:
I changed it to prompt_tokens instead of max_tokens because we don't actually tell it to give us up to N tokens of context; we tell it to include all of the user's include files (since we no longer truncate it) and give us up to auto_tokens of automatic context (as long as it fits). The prompt tokens is just so we can check if there are enough tokens for user included files, and so we can only add as many auto context files as can fit in context. I think this makes a lot more sense to me (since we aren't truncating user included files anymore).

Reply:
What do you mean by difficult to use with different models/scenarios? I'm not sure I understand